lib/ice_cube/validations/monthly_by_set_pos.rb

```diff
@@ -0,0 +1,76 @@
+module IceCube
+
+  module Validations::MonthlyBySetPos
+    def by_set_pos(*by_set_pos)
+      by_set_pos.flatten!
+      by_set_pos.each do |set_pos|
+        unless (-366..366).include?(set_pos) && set_pos != 0
+          raise ArgumentError, "Expecting number in [-366, -1] or [1, 366], got #{set_pos} (#{by_set_pos})"
+        end
+      end
+
+      @by_set_pos = by_set_pos
+      replace_validations_for(:by_set_pos, [Validation.new(by_set_pos, self)])
+      self
+    end
+
+    class Validation
+
+      attr_reader :rule, :by_set_pos
+
+      def initialize(by_set_pos, rule)
+        @by_set_pos = by_set_pos
+        @rule = rule
+      end
+
+      def type
+        :day
+      end
+
+      def dst_adjust?
+        true
+      end
+
+      def validate(step_time, start_time)
+        start_of_month = step_time.beginning_of_month
+        end_of_month = step_time.end_of_month
```

Comment:
Without active support, this might work:

```ruby
last_day = IceCube::TimeUtil.days_in_month(step_time)
end_of_month = IceCube::TimeUtil.build_in_zone([step_time.year, step_time.mont
h, last_day, 23, 59, 59], step_time)
```